operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java

```diff
@@ -0,0 +1,102 @@
+package org.keycloak.operator;
+
+import io.quarkus.logging.Log;
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.v2alpha1.KeycloakService;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition;
+import org.keycloak.operator.utils.K8sUtils;
+
+import java.time.Duration;
+import java.util.List;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.Constants.KEYCLOAK_DISCOVERY_SERVICE_PORT;
+import static org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition.READY;
+
+@QuarkusTest
+public class ClusteringE2EIT extends ClusterOperatorTest {
```

Comment:
We're missing any test for the discovery service. This should be ideally an Infinispan test that would check the nodes really see each other, but I'm not sure how to do that and if it's even possible. Maybe @hmlnarik might have some tips?

Reply:
The test `testKeycloakCacheIsConnected` prove that Infinispan is connected.